### PR TITLE
Fix source seed selection docs generate

### LINKED ISSUE #9161: [CT-3429] [Bug] Selectors don't apply for sources and seeds in `dbt docs generate`
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

Following up from https://github.com/dbt-labs/dbt-core/pull/9159, if there are multiple seeds or sources in a dbt project and a selector is used, `catalog.json` should only contain the seeds or sources that match selection criteria (which is currently ignored).

### Expected Behavior

The selection criteria above are respected.

### Steps To Reproduce

1. Add seeds `my_first_seed` and `my_second_seed` to project and run `dbt seed`
2. Run `dbt docs generate --select my_first_seed`
3. Inspect `catalog.json` to find that `my_second_seed` is in there even though it shouldn't be

### Relevant log output

_No response_

### Environment

```markdown
- OS: MacOS (applies to all)
- Python: 3.11
- dbt: 1.7
- adapter: Postgres, but probably applies to all
```

### Which database adapter are you using with dbt?

postgres

### Additional Context

Acceptance Criteria:
1. Fix TODOs here: https://github.com/dbt-labs/dbt-core/blob/201d75bd1c3d8da7225990f96ccfe8987dcd70a8/tests/functional/docs/test_generate.py#L90-L100
2. Document other node types that selectors haven't been tested against

### PR DESCRIPTION
**Pull request by MichelleArk**
resolves #9161

### Problem

1. Sources are not actively participating in resource selection in docs generate. They cannot simply be included as nodes in the ResourceTypeSelector because then the task would attempt to compile a SourceDefinition which leads to all sorts of errors since they are not executable. 

2. Seed selection actually does work, but in the edge case where a seed and source have the _same_ database identifier, and the seed is selected, its catalog metadata is populated under `catalog.sources` instead of in `catalog.nodes`. Vice versa happens if the source is selected but the seed is not. If both are selected, both are in the catalog as expected (already tested).

### Solution

1. Use the `ResourceTypeSelector` with `SourceDefinition` during catalog generation, after compilation to include queried sources in the catalog result. 

2. Handle edge case by accepting the selected_ids to `catalog.make_unique_id_map` to only create node and source maps for selected seeds. Update the seed selection test to explicitly test selection for a seed that shares a relation with a source.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [x] I have run this code in development and it appears to resolve the stated issue  
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX
- [x] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

### CODE REVIEW

File: core/dbt/task/docs/generate.py [hunk 1]
```diff
@@ -251,14 +261,19 @@ def run(self) -> CatalogArtifact:
...
+                    # Source selection is handled separately from main job_queue selection because
+                    # SourceDefinition nodes cannot be safely compiled / run by the CompileRunner / CompileTask,
+                    # but should still be included in the catalog based on the selection spec
+                    selected_source_ids = self._get_selected_source_ids()
+                    selected_source_nodes = self._get_nodes_from_ids(
+                        self.manifest, selected_source_ids
                     )
-                    selected_nodes.extend(source_ids)
+                    selected_node_ids.update(selected_source_ids)
+                    selected_nodes.extend(selected_source_nodes)
```
[the remark this answers is not in the document]
Comment (MichelleArk) [reply core/dbt/task/docs/generate.py:272]: this led to some unsavory mypy issues, so I left it as-is.
